docs/rst/use_cases/keys/keys.rst

```diff
@@ -0,0 +1,183 @@
+====================================
+
+.. contents::
+    :depth: 2
+    :local:
+    :backlinks: none
+
+Background
+----------
+
+This use case explains how to run three turtlesim nodes by leveraging topic keys and content filtering for efficient communication.
+
+* **Topic keys** reduce the number of entities needed in the data flow, leading to more efficient resource and bandwidth usage.
+* **Content Filter Topics** allow subscribers to receive only the messages that match specific criteria, ensuring each Turtle Node processes only relevant data.
+
+By integrating topic keys and content filtering, each turtle node processes the data intended for it.
+This setup illustrates efficient and effective communication in a complex system with multiple entities.
+
+Users can control the turtles using the keyboard, while the Qt application visually represents their movements.
+The Qt application terminal displays the turtles' positions, and the controller terminal shows the commands sent to each Turtle Node.
+The controller receives the turtles position.
+Each velocity and pose message is tagged with a key representing the turtle ID.
+
+The diagram below shows the entities required using topic keys:
+
+.. figure:: /rst/figures/use_cases/keys/keys.png
+    :align: center
+
+The diagram below shows the entities required **without** using topic keys:
+
+.. figure:: /rst/figures/use_cases/keys/non-keys.png
+    :align: center
+
+.. note::
+
+    Topic keys work exclusively with publishers and subscribers, not with services.
+    Multiple service clients can use the same service, but there can only be one service server for a given service.
+    Since each Turtle Node must have a service server to control goals, it does not make sense to have multiple clients (e.g., controllers) using the same service server with different keys.
+
+
+Prerequisites
+-------------
+
+Familiarity with the following topics is recommended:
+
+* :ref:`topic_keys`
+* `Content Filter Topic <https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/topic/contentFilteredTopic/contentFilteredTopic.html>`__
+
+It is required to have previously installed Vulcanexus using one of the following installation methods:
+
+* :ref:`linux_binary_installation`
+* :ref:`linux_source_installation`
+* :ref:`docker_installation`
+
+
+Keyed Messages
+--------------
+
+The keyed Topics use the following message types to represent the turtles' velocity and pose:
+
+``docs_turtlesim::msg::KeyedPose`` Represents the pose of a turtle with a key.
+
+- `key:` Identifier of the turtle.
+- `x:` X-coordinate of the turtle's position.
+- `y:` Y-coordinate of the turtle's position.
+- `theta:` Orientation angle of the turtle.
+- `linear_velocity:` Linear velocity of the turtle.
+- `angular_velocity:` Angular velocity of the turtle.
+
+.. code-block:: bash
+
+    // KeyedPose.idl
+    module docs_turtlesim {
+        module msg {
+            struct KeyedPose {
+                @key long key;
```

Comment:
Do not name `key` to the keyed member just because its the key of the message. 
```suggestion
                @key long turtle_id; // identifier of the turtle.
```